### PR TITLE
[ avcpp ] Update version

### PR DESCRIPTION
**Pull request by barcharcraz**
updates avcpp. They accepted the "remove problematic compound literal" patch, so we can delete it from vcpkg.

### CODE REVIEW

File: ports/avcpp/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "avcpp",
-  "version-date": "2021-05-06",
-  "port-version": 1,
+  "version-date": "2021-06-14",
+  "port-version": 2,
```
Comment (Hoikas) [inline ports/avcpp/vcpkg.json:4]:
```suggestion
```

When the `version-date` is updated, the `port-version` resets to zero.

**Comment by barcharcraz** [comment]
reverted the port version and updated the vcpkg versions.

**Comment by JonLiu1993** [comment]
@barcharcraz ,Please add "vcpkg_cmake_pkgconfig()" in the portfile.cmake

**Comment by JonLiu1993** [comment]
@barcharcraz ，After modifying the content I suggest, we can merge this pr

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks!

**Pull request merged**